pkg/sdk/grants_impl.go

```diff
@@ -344,6 +350,98 @@ func (v *grants) grantOwnershipOnPipe(ctx context.Context, pipeId SchemaObjectId
 	return nil
 }
 
+func (v *grants) grantOwnershipOnTask(ctx context.Context, taskId SchemaObjectIdentifier, opts *GrantOwnershipOptions) error {
+	currentGrantsOnObject, err := v.client.Grants.Show(ctx, &ShowGrantOptions{
+		On: &ShowGrantsOn{
+			Object: &Object{
+				ObjectType: ObjectTypeTask,
+				Name:       taskId,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	currentGrantsOnAccount, err := v.client.Grants.Show(ctx, &ShowGrantOptions{
+		On: &ShowGrantsOn{
+			Account: Bool(true),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	currentRole, err := v.client.ContextFunctions.CurrentRole(ctx)
+	if err != nil {
+		return err
+	}
+	currentRoleName := NewAccountObjectIdentifier(currentRole)
+
+	currentTask, err := v.client.Tasks.ShowByID(ctx, taskId)
+	if err != nil {
+		return err
+	}
+
+	currentGrantsOnTaskWarehouse, err := v.client.Grants.Show(ctx, &ShowGrantOptions{
+		On: &ShowGrantsOn{
+			Object: &Object{
+				ObjectType: ObjectTypeWarehouse,
+				Name:       NewAccountObjectIdentifier(currentTask.Warehouse),
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	isGrantedWithPrivilege := func(collection []Grant, grantedOn ObjectType, privilege string) bool {
+		return slices.ContainsFunc(collection, func(grant Grant) bool {
+			return grant.GranteeName == currentRoleName &&
+				grant.GrantedOn == grantedOn &&
+				grant.Privilege == privilege
+		})
+	}
+	canOperateOnTask := isGrantedWithPrivilege(currentGrantsOnObject, ObjectTypeTask, SchemaObjectPrivilegeOperate.String())
+	isGrantedWithWarehouseUsage := isGrantedWithPrivilege(currentGrantsOnTaskWarehouse, ObjectTypeWarehouse, AccountObjectPrivilegeUsage.String())
+	canSuspendTask := canOperateOnTask || isGrantedWithPrivilege(currentGrantsOnObject, ObjectTypeTask, "OWNERSHIP")
+	canResumeTask := isGrantedWithWarehouseUsage || canOperateOnTask || isGrantedWithPrivilege(currentGrantsOnAccount, ObjectTypeAccount, GlobalPrivilegeExecuteTask.String())
+
+	var tasksToResume []SchemaObjectIdentifier
+	if canSuspendTask {
+		tasksToResume, err = v.client.Tasks.SuspendRootTasks(ctx, taskId, taskId)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Printf("[WARN] Insufficient privileges to operate on task: %s (OPERATE privilege). Trying to proceed with ownership transfer...", taskId.FullyQualifiedName())
+	}
+
+	if err := validateAndExec(v.client, ctx, opts); err != nil {
+		return err
+	}
+
+	if currentTask.State == TaskStateStarted && !slices.ContainsFunc(tasksToResume, func(id SchemaObjectIdentifier) bool {
+		return id.FullyQualifiedName() == currentTask.ID().FullyQualifiedName()
+	}) {
+		tasksToResume = append(tasksToResume, currentTask.ID())
+	}
+
+	if len(tasksToResume) > 0 {
+		if canResumeTask && ((opts.CurrentGrants != nil && opts.CurrentGrants.OutboundPrivileges == Copy) || (opts.To.AccountRoleName != nil && opts.To.AccountRoleName.Name() == currentRoleName.Name())) {
```

Comment:
can we extract `((opts.CurrentGrants != nil && opts.CurrentGrants.OutboundPrivileges == Copy) || (opts.To.AccountRoleName != nil && opts.To.AccountRoleName.Name() == currentRoleName.Name()))` to function(s), too? The rest of the conditionals have more readable conditions